### PR TITLE
K8s backup

### PR DESCRIPTION
**Pull request by kostas-kou**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
Some minor changes for making the backup working with postgres tls
and logs have been added

### Changes made:
1) Add the `verify-full` ssl mode which was missing
2) Change the `sslmode` env var to avoid any mistakes during the viper translation
3) Add more logs for postgres, s3 etc

### Additional information:
This PR is part of the issue: NBISweden/BigPicture-Deployment#133

### CODE REVIEW

**Review by aaperis — APPROVED** [review]

File: postgres.go [hunk 1]
```diff
@@ -308,6 +308,8 @@ func buildConnInfo(db DBConf) string {
 		certsRequired = true
 	case "verify-peer":
 		certsRequired = true
+	case "verify-full":
+		certsRequired = true
```
Comment (jbygdell) [inline postgres.go:312]: This should no be an addition, it should replace ` verify-peer`

Comment (kostas-kou) [reply postgres.go]: Right, it is done now

File: mongo.go [hunk 2]
```diff
@@ -40,37 +41,44 @@ func (mongo mongoConfig) dump(sb s3Backend, keyPath, database string) error {
...
+	mongoArchive := today + "-" + database + ".archive"
+	wr, err := sb.NewFileWriter(mongoArchive, &wg)
 	if err != nil {
-		log.Errorf("Could not open backup file for writing: %v", err)
+		log.Error("Could not open backup file for writing")
 
 		return err
 	}
 
+	log.Debugf("Mongo archive file %v ready for writting", mongoArchive)
```
Comment (norling) [inline mongo.go:58]:
```suggestion
	log.Debugf("Mongo archive file %v ready for writing", mongoArchive)
```

File: postgres.go [hunk 3]
```diff
@@ -44,57 +45,63 @@ func (db DBConf) basebackup(sb s3Backend, keyPath string) error {
...
 	wg := sync.WaitGroup{}
 	wr, err := sb.NewFileWriter(fileName, &wg)
 	if err != nil {
-		log.Errorf("Could not open backup file for writing: %v", err)
+		log.Error("Could not open backup file for writing")
 
 		return err
 	}
 
+	log.Debugf("Backup file %v ready for writting", fileName)
```
Comment (norling) [inline postgres.go:84]:
```suggestion
	log.Debugf("Backup file %v ready for writing", fileName)
```

File: postgres.go [hunk 4]
```diff
@@ -132,37 +142,44 @@ func (db DBConf) dump(sb s3Backend, keyPath string) error {
...
+	dumpFile := today + "-" + db.database + ".sqldump"
+	wr, err := sb.NewFileWriter(dumpFile, &wg)
 	if err != nil {
-		log.Errorf("Could not open backup file for writing: %v", err)
+		log.Error("Could not open backup file for writing")
 
 		return err
 	}
 
+	log.Debugf("Dump file %v ready for writting", dumpFile)
```
Comment (norling) [inline postgres.go:159]:
```suggestion
	log.Debugf("Dump file %v ready for writing", dumpFile)
```

**Review by norling — APPROVED** [review]
Some typos! Also - the README still references `verify-peer` which I think should be changed to `verify-full`.

File: postgres.go [hunk 5]
```diff
@@ -44,57 +45,63 @@ func (db DBConf) basebackup(sb s3Backend, keyPath string) error {
 
 	err := cmd.Run()
 	if err != nil {
-		log.Errorf(errMsg.String())
-
```
Comment (jbygdell) [inline postgres.go:48]: The kinds of code blocks can be written in a cleaner way:
```
if err := cmd.Run(); err != nil {
	retrurn err
}
```

**Review by norling — APPROVED** [review]

**Review by aaperis — APPROVED** [review]

**Pull request merged**